cristin-organization/src/main/java/no/unit/nva/cristin/organization/CristinApiClient.java

```diff
@@ -45,15 +52,74 @@ public Organization getOrganization(URI uri)
+                .getUri();
+
+        long start = System.currentTimeMillis();
+        SearchResponse<Organization> searchResponse = httpExecutor.query(queryUri);
+        final long totalProcessingTime = System.currentTimeMillis() - start;
+        return updateSearchResponseMetadata(searchResponse,requestQueryParams, totalProcessingTime);
+    }
+
+    private Map<String, String> translateToCristinApi(Map<String, String> requestQueryParams) {
+        return Map.of(
+                CRISTIN_QUERY_NAME_PARAM, requestQueryParams.get(QUERY),
+                PAGE, requestQueryParams.get(PAGE),
+                CRISTIN_PER_PAGE_PARAM, requestQueryParams.get(NUMBER_OF_RESULTS));
+    }
+
+    private SearchResponse<Organization> updateSearchResponseMetadata(
+            SearchResponse<Organization> searchResponse,
+            Map<String, String> requestQueryParams,
+            long timeUsed) {
+        searchResponse.setId(getNvaQueryId(requestQueryParams));
+        searchResponse.setFirstRecord(calculateFirstRecord(requestQueryParams));
+        searchResponse.setNextResults(nextResult(getNvaApiBaseUri(), requestQueryParams, searchResponse.getSize()));
+        searchResponse.setPreviousResults(previousResult(getNvaApiBaseUri(),
+                requestQueryParams,
+                searchResponse.getSize()));
+        searchResponse.setProcessingTime(timeUsed);
+        return searchResponse;
+    }
+
+    private URI getNvaQueryId(Map<String, String> requestQueryParams) {
+        return new UriWrapper(getNvaApiBaseUri()).addQueryParameters(requestQueryParams).getUri();
+    }
+
+    private URI getNvaApiBaseUri() {
+        return new UriWrapper(HTTPS, DOMAIN_NAME).addChild(BASE_PATH).addChild(ORGANIZATION_PATH).getUri();
+    }
+
+    private URI previousResult(URI baseUri, Map<String, String> requestQueryParams, int totalSize) {
+        int firstPage = Integer.parseInt(requestQueryParams.get(PAGE)) - 1;
+        if (firstPage > 0 && totalSize > 0) {
+            Map<String, String> nextMap = new ConcurrentHashMap<>(requestQueryParams);
+            nextMap.put(PAGE, Integer.toString(firstPage));
+            return new UriWrapper(baseUri).addQueryParameters(nextMap).getUri();
+        }
+        return null;
+    }
+
+    private URI nextResult(URI baseUri, Map<String, String> requestQueryParams, int totalSize) {
+        int currentPage = Integer.parseInt(requestQueryParams.get(PAGE));
+        int pageSize = Integer.parseInt(requestQueryParams.get(NUMBER_OF_RESULTS));
+        if (currentPage * pageSize < totalSize) {
+            Map<String, String> nextMap = new ConcurrentHashMap<>(requestQueryParams);
+            nextMap.put(PAGE, Integer.toString(currentPage + 1));
+            return new UriWrapper(baseUri).addQueryParameters(nextMap).getUri();
+        }
+        return null;
+    }
+
+    private Integer calculateFirstRecord(Map<String, String> requestQueryParams) {
```

Comment:
This is now in both this class and SearchResponse class. Maybe we should reuse the one, `indexOfFirstEntryInPageCalculatedFromParams` in SearchResponse by making it public?

Reply:
Or the other way around, reuse you method in SearchResponse, but then it have to be moved to some common class

Reply:
I think since this logic is used in SearchResponse the logic belongs to that class. For example a static functional method in SearchResponse which can be used everywhere and does not rely on state

Reply:
What if there are zero results? `(1 - 1) * 5 + 1 = 1`

Reply:
Checking if we have hits